api/server/router/local/image.go

```diff
@@ -56,13 +61,18 @@ func (s *router) postCommit(ctx context.Context, w http.ResponseWriter, r *http.
 		Config:  c,
 	}
 
-	imgID, err := builder.Commit(cname, s.daemon, commitCfg)
+	container, err := s.daemon.Get(cname)
```

Comment:
@LK4D4 the API is still dependent on daemon unfortunately. I think it's still valuable to have this first PR and have another one remove completely daemon dependency. It's already a pretty big PR :S